core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -392,15 +396,22 @@ public IndicesQueriesRegistry buildQueryParserRegistry() {
 
         for (QueryRegistration<?> query : queries) {
             QueryParser<? extends QueryBuilder<?>> parser = query.parser;
-            for (String name: query.names) {
+            for (String name: query.parseField.getAllNamesIncludedDeprecated()) {
```

Comment:
I dropped QueryRegistration in https://github.com/elastic/elasticsearch/pull/17458.